Dockerfile

```diff
@@ -7,7 +7,7 @@ ARG RUST_IMAGE=ghcr.io/linkerd/dev:v48-rust
 
 # Use an arbitrary ~recent edge release image to get the proxy
 # identity-initializing and linkerd-await wrappers.
-ARG LINKERD2_IMAGE=ghcr.io/linkerd/proxy:edge-23.11.2
+ARG LINKERD2_IMAGE=ghcr.io/linkerd/proxy:edge-25.11.3
```

Comment:
would pointing this at `:latest` be even better? this drifted more than two years out of date. luckily there was only one other breaking change to spot and fix, but i'd like for this repo to not get so far out of step with the linkerd2 repo again.